brendanarnold/py-fortranformat
# Poetry install and GitHub actions

Adds a `pyproject.toml` based build using a `pyproject` based build

To install
```
python -m venv .venv
source ./.venv/bin/activate
python -m pip install --upgrade pip
pip install '.[test,build]'
pytest tests
```

This PR also adds Github Actions to automatically build and test the library, and to push tags to pypi.org. For this to work, the repository owner of https://pypi.org/project/fortranformat/ will need to add Github Actions as a trusted publisher. This basically means
1. Go to https://pypi.org/manage/project/fortranformat/settings/publishing/
2. On the Github tab, add owner: `brendanarnold`, repository: `py-fortranformat`, workflow name: `workflow_actions.yml`, environment name: `pypi-publish` and click Add. See [docs](https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/configuring-openid-connect-in-pypi)
3. Go to https://github.com/brendanarnold/py-fortranformat/settings/environments. Click `new environment` and add an environment called `pypi-publish` and add maintainers as required reviewers. See [docs](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-deployments/managing-environments-for-deployment)
4. Make a new tag. This will automatically run the Github Action and then ask if you want to push to pypi
